stactask/task.py

```diff
@@ -218,13 +225,10 @@ def cleanup_workdir(self) -> None:
             )
 
     def assign_collections(self) -> None:
-        """Assigns new collection names based on"""
-        for i, (coll, expr) in itertools.product(
-            self._payload["features"],
-            self.upload_options.get("collections", dict()).items(),
-        ):
-            if stac_jsonpath_match(i, expr):
-                i["collection"] = coll
+        """Assigns new collection names based on upload_options collections attribute"""
+        for item in self._payload["features"]:
+            if coll := utils_find_collection(self.collection_mapping, item):
+                item["collection"] = coll
```

Comment:
Nice.  I like that this gets out of checking against _all_ the collections, and just yields the first match

Reply:
Should it be emphasized that `in the event of multiple matches, python's dict ordering will determine which collection is used`.  Or perhaps this is obvious enough.